src/mrpro/algorithms/reconstruction/TotalVariationRegularizedReconstruction.py

```diff
@@ -0,0 +1,142 @@
+"""Total Variation (TV)-Regularized Reconstruction using PDHG."""
+
+from __future__ import annotations
+
+from collections.abc import Callable, Sequence
+
+import torch
+
+from mrpro.algorithms.optimizers.pdhg import pdhg
+from mrpro.algorithms.prewhiten_kspace import prewhiten_kspace
+from mrpro.algorithms.reconstruction.DirectReconstruction import DirectReconstruction
+from mrpro.data.CsmData import CsmData
+from mrpro.data.DcfData import DcfData
+from mrpro.data.IData import IData
+from mrpro.data.KData import KData
+from mrpro.data.KNoise import KNoise
+from mrpro.operators import LinearOperatorMatrix, ProximableFunctionalSeparableSum
+from mrpro.operators.FiniteDifferenceOp import FiniteDifferenceOp
+from mrpro.operators.functionals import L1NormViewAsReal, L2NormSquared, ZeroFunctional
+from mrpro.operators.LinearOperator import LinearOperator
+from mrpro.utils import unsqueeze_right
+
+
+class TotalVariationRegularizedReconstruction(DirectReconstruction):
+    r"""TV-regularized reconstruction.
+
+    This algorithm solves the problem :math:`min_x \frac{1}{2}||Ax - y||_2^2 + \sum_i l_i ||\nabla_i x||_1`
+    by using the PDHG-algorithm. :math:`A` is the acquisition model (coil sensitivity maps, Fourier operator,
+    k-space sampling), :math:`y` is the acquired k-space data, :math:`l_i` are the strengths of the regularization
+    along the different dimensions and :math:`\nabla_i` is the finite difference operator applied to :math:`x` along
+    different dimensions :math:`i`.
+    """
+
+    max_iterations: int
+    """Maximum number of PDHG iterations."""
+
+    tolerance: float
+    """Tolerance of PDHG for relative change of the primal solution."""
+
+    regularization_weights: torch.Tensor
+    """Strengths of the regularization along different dimensions :math:`l_i`."""
+
+    def __init__(
+        self,
+        kdata: KData | None = None,
+        fourier_op: LinearOperator | None = None,
+        csm: Callable | CsmData | None = CsmData.from_idata_walsh,
+        noise: KNoise | None = None,
+        dcf: DcfData | None = None,
+        *,
+        max_iterations: int = 100,
+        tolerance: float = 0,
+        regularization_weights: Sequence[float] | Sequence[torch.Tensor],
+    ) -> None:
+        """Initialize TotalVariationRegularizedReconstruction.
+
+        Parameters
+        ----------
+        kdata
+            KData. If kdata is provided and fourier_op or dcf are None, then fourier_op and dcf are estimated based on
+            kdata. Otherwise fourier_op and dcf are used as provided.
+        fourier_op
+            Instance of the FourierOperator used for reconstruction. If None, set up based on kdata.
```

Comment:
```suggestion
            Instance of the FourierOperator used for reconstruction. If None, set up based on `kdata`.
```

Reply:
(I think this would be enough to make FourierOperator linked in docs, if not, we must use the full name)

Reply:
could you also do this for the other references?